WordPress/src/main/java/org/wordpress/android/ui/posts/services/PostUploadService.java

```diff
@@ -629,8 +676,53 @@ public void onPostUploaded(OnPostUploaded event) {
     @SuppressWarnings("unused")
     @Subscribe(threadMode = ThreadMode.MAIN)
     public void onMediaUploaded(OnMediaUploaded event) {
+        if (event.media == null) {
+            return;
+        }
+        if (!mUseLegacyMode) {
+            handleMediaUploadCompleted(event);
+        } else {
+            handleMediaUploadCompletedLegacy(event);
+        }
+    }
+
+    private void handleMediaUploadCompleted(OnMediaUploaded event) {
+        if (event.media.getLocalPostId() == 0) {
+            // Only interested in media attached to local posts
+            return;
+        }
+
+        if (event.isError()) {
+            // TODO: Find the associated post, mark it as failed, update upload messaging, and remove it from the queue
+            AppLog.e(T.MEDIA, "Media upload failed for post " + event.media.getLocalPostId() + " : " +
+                    event.error.type + ": " + event.error.message);
+
+            PostModel postToCancel = removeQueuedPostByLocalId(event.media.getLocalPostId());
+            if (postToCancel == null) return;
+
+            mFirstPublishPosts.remove(postToCancel.getId());
+            EventBus.getDefault().post(new PostEvents.PostUploadCanceled(postToCancel.getLocalSiteId()));
+            finishUpload();
+            return;
+        }
+
+        if (event.canceled) {
+            // TODO: If a media upload for a post was cancelled, we might want to cancel the post upload, too
+            // Not implemented
```

Comment:
Hmm, yeah okay. The reason I added a TODO was so we could think out this use case a bit more - currently there's no way for this to be called (can't cancel a media upload outside the editor, and re-opening a post with uploading media will cancel the post upload, so cancelling at that point will also not call this).

But I think it makes sense to just add the behavior with some logging so we can tell this is happening if we're confused about something in the future.